### PR TITLE
#623 - Harden delete vPool

### PR DESCRIPTION
**Pull request by kvanhijf**
Alba proxy configuration key always removed now
Partitions shown as failed when related disk is in error or missing state
Saving disk during sync with reality should always succeed now

### CODE REVIEW

File: ovs/lib/storagerouter.py [hunk 1]
```diff
@@ -1063,97 +1056,85 @@ def remove_storagedriver(storagedriver_guid, offline_storage_router_guids=None):
...
-                            except Exception as ex:  # We don't put errors_found to True, because ensure safety could possibly succeed later on
-                                StorageRouterController._logger.error('Remove Storage Driver - Guid {0} - Virtual Disk {1} {2} - Ensuring MDS safety failed with error: {3}'.format(storage_driver.guid, vdisk.guid, vdisk.name, ex))
-                    else:
-                        StorageRouterController._logger.info('Remove Storage Driver - Guid {0} - Virtual Disk {1} {2} - No migration required'.format(storage_driver.guid, vdisk.guid, vdisk.name))
+        available_sr_names = [sd.storagerouter.name for sd in available_storage_drivers]
+        StorageRouterController._logger.info('Remove Storage Driver - Guid {0} - Storage Routers on which an available Storage Driver runs: {1}'.format(storage_driver.guid, ', '.join(available_sr_names)))
+
+        # Remove stale vDisks
+        voldrv_vdisks = [entry.object_id() for entry in vpool.objectregistry_client.get_all_registrations()]
+        model_vdisk_guids = [vdisk.guid for vdisk in VDiskList.get_vdisks() if vdisk.vpool == vpool]
```
Comment (khenderick) [inline ovs/lib/storagerouter.py]: You can use `vpool.vdisks` here

File: ovs/lib/storagerouter.py [hunk 2]
```diff
@@ -1329,38 +1312,42 @@ def remove_storagedriver(storagedriver_guid, offline_storage_router_guids=None):
...
             StorageRouterController._logger.info('Remove Storage Driver - Guid {0} - Checking DTL for all virtual disks in vPool {1} with guid {2}'.format(storage_driver.guid, vpool.name, vpool.guid))
             try:
                 VDiskController.dtl_checkup(vpool_guid=vpool.guid, ensure_single_timeout=600)
-            except Exception as ex:
-                StorageRouterController._logger.error('Remove Storage Driver - Guid {0} - DTL checkup failed for vPool {1} with guid {2} with error: {3}'.format(storage_driver.guid, vpool.name, vpool.guid, ex))
+            except Exception:
+                StorageRouterController._logger.exception('Remove Storage Driver - Guid {0} - DTL checkup failed for vPool {1} with guid {2}'.format(storage_driver.guid, vpool.name, vpool.guid))
+
+        if sd_can_be_deleted is True:
+            storage_driver.delete(abandon=['logs'])  # Detach from the log entries
```
Comment (khenderick) [inline ovs/lib/storagerouter.py]: Logs isn't required anymore, you can clean that up

File: webapps/api/backend/views/vpools.py [hunk 3]
```diff
@@ -99,7 +99,7 @@ def shrink_vpool(self, vpool, storagerouter_guid):
         if sd_guid is None:
             raise HttpNotAcceptableException(error_description='Storage Router {0} is not a member of vPool {1}'.format(sr.name, vpool.name),
                                              error='impossible_request')
-        return StorageRouterController.remove_storagedriver.s(sd_guid).apply_async(routing_key='sr.{0}'.format(sr.machine_id))
+        return StorageRouterController.remove_storagedriver.s(sd_guid).apply_async()
```
Comment (khenderick) [inline webapps/api/backend/views/vpools.py]: Use `.delay()` here, that's easier on the eyes.

**Review by khenderick — CHANGES_REQUESTED** [review]

**Review by khenderick — APPROVED** [review]

**Comment by kvanhijf** [comment]
openvstorage-2.7.3-rev.3978.4f93e02

**Pull request merged**